keras_cv/models/__internal__/darknet_utils.py

```diff
@@ -0,0 +1,398 @@
+    """
+
+    def __init__(
+        self, filters, kernel_size, strides, use_bias=False, activation="silu", **kwargs
+    ):
+        super().__init__(**kwargs)
+
+        self.filters = filters
+        self.kernel_size = kernel_size
+        self.strides = strides
+        self.use_bias = use_bias
+        self.activation = activation
+
+        self.conv = layers.Conv2D(
+            filters,
+            kernel_size,
+            strides,
+            padding="same",
+            use_bias=use_bias,
+        )
+
+        self.batch_norm = layers.BatchNormalization()
+
+        if activation == "silu":
+            self.act = layers.Lambda(lambda x: keras.activations.swish(x))
+        elif activation == "relu":
+            self.act = layers.ReLU()
+        elif activation == "leaky_relu":
+            self.act = layers.LeakyReLU(0.1)
+
+    def call(self, x):
+        x = self.conv(x)
+        x = self.batch_norm(x)
+        x = self.act(x)
+        return x
+
+    def get_config(self):
+        config = {
+            "filters": self.filters,
+            "kernel_size": self.kernel_size,
+            "strides": self.strides,
+            "use_bias": self.use_bias,
+            "activation": self.activation,
+        }
+        base_config = super().get_config()
+        return dict(list(base_config.items()) + list(config.items()))
+
+
+def ResidualBlocks(filters, num_blocks, name=None):
+    """A residual block used in DarkNet models, repeated `num_blocks` times.
+
+    Args:
+        filters: Integer, the dimensionality of the output spaces (i.e. the number of
+            output filters in used the blocks).
+        num_blocks: number of times the residual connections are repeated
+        name: the prefix for the layer names used in the block.
+
+    Returns:
+        a function that takes an input Tensor representing a ResidualBlock.
+    """
+
+    if name is None:
+        name = f"residual_block{backend.get_uid('residual_block')}"
+
+    def apply(x):
+        x = DarknetConvBlock(
+            filters,
+            kernel_size=3,
+            strides=2,
+            activation="leaky_relu",
+            name=f"{name}_conv1",
+        )(x)
+
+        for i in range(1, num_blocks + 1):
+            residual = x
+
+            x = DarknetConvBlock(
+                filters // 2,
+                kernel_size=1,
+                strides=1,
+                activation="leaky_relu",
+                name=f"{name}_conv{2*i}",
+            )(x)
+            x = DarknetConvBlock(
+                filters,
+                kernel_size=3,
+                strides=1,
+                activation="leaky_relu",
+                name=f"{name}_conv{2*i + 1}",
+            )(x)
+
+            if i == num_blocks:
+                x = layers.Add(name=f"{name}_out")([residual, x])
+            else:
+                x = layers.Add(name=f"{name}_add_{i}")([residual, x])
+
+        return x
+
+    return apply
+
+
+def SPPBottleneck(
```

Comment:
lets not abbreviate SPP

Reply:
`SpatialPyramidPoolingBottleneck` sounded a bit too long. I assumed this abbreviation made it more readable.

Should I change it to `SpatialPyramidPoolingBottleneck`?

Reply:
yeah, please do.  This is not a user facing API so a longer descriptive name is better